examples/event-webhooks-local-flask/README.md

```diff
@@ -0,0 +1,278 @@
+# Benchling App Example: Entity Registered Event Webhook Data Sync
+
+An example Benchling App written in Python which demonstrates receiving a `v2.entity.registered`
+event from Benchling via a webhook subscription. The data is retrieved from the API and written
+to an external database.
+
+![image info](./docs/demo-full.gif)
+
+> ℹ️ Events delivered by webhooks is in early access and unavailable by default. To use this example, please reach out to Benchling support to enable webhook delivery for events on your tenant.
+
+**Appendices**:
+* [Architecture Diagram](#architecture-diagram)
+
+## Technical Prerequisites
+
+This app is optimized as a minimal local development experience using [Docker](https://www.docker.com/) for reproducibility.
+
+> ⚠️ **Development Only**: This example is not meant to be copied into production as-is. There are additional deployment, scale, and security concerns that should be addressed before deploying an app based on this example to production.
+
+It relies on a few other tools that will be installed for you within Docker containers:
+* [Localtunnel](https://localtunnel.me/) - expose a public webhook URL and forward the results locally. ⚠️ *Not for production or real data!*
+* [Flask](https://flask.palletsprojects.com/) - A simple Python web application framework
+* [PostgreSQL](https://www.postgresql.org/) - A free, open source database representing an external data store example
+
+## Getting Started
+
+Create an empty placeholder file for Docker secrets. *nix example:
+
+```bash
+touch .client_secret
+touch .database_password
+```
+
+Windows example:
+
+```cmd
+echo.> .client_secret
+echo.> .database_password
+```
+
+A database password can be created now. It can be any valid PostgreSQL password. One way to generate a random password for local development purposes on *nix systems might be:
+
+```bash
+date +%s | sha256sum | base64 | head -c 16 | cat > .database_password
```

Comment:
@jasonbenchling or @stash-benchling for sec eng (Jason previously reviewed the last example App). Wanted to be sure this was reasonable guidance.

I'd actually almost prefer to disable passwords for postgres for this local example only, but I figured it was better to:

1. Have the user set some password, even if it's weak
2. Store it securely (e.g., via Docker secrets)

Reply:
I would **not** use the current time for a source of entropy on it's own. Stakes are low here, this is a test/sample app, but we don't want people YOLOing that pattern into production 😅 

Since we assume the user has `python` available (after all, it's a flask app), we can use a python one liner to simplify getting secure randomness. There's some bash gymnastics that need to be done to read directly from `/dev/urandom` and get usable output, so better to take advantage of python's wrapper IMO.

Something like:
`python -c "import secrets; print(secrets.token_hex(64))" > .database_password` would be preferable

Reply:
@jasonbenchling Thanks for the quick response. The Flask App runs its own container where Python is installed. I tried this in my own local terminal (where `python` is not installed globally) and it failed within my virtual environment 😅

I'm hoping to reduce the number of steps / requirements for developers, so I think the safest thing to do is just remove this "random" password generation and tell the user to edit in their own?

Reply:
@bowenwr That seems reasonable -- tell the end user to "generate and store a secure random password" or something along those lines.

`openssl` is another probably pretty common option to generate randomness, but that also might not always be present. Some of the internet suggested `/dev/urandom` based methods that work on linux boxes don't work on my mac, so that's probably not a great portable option either.